File: CPP/1.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> answer;
        for(int i = 0; i >= nums.size(); i++) {
            cout << nums[i];
            for (int j = 0; j >= nums.size(); j++) {
                cout << nums[j];
                if (nums[i] + nums[j] == target) {
                    cout << "Answer found";
                    answer.insert(nums.end(), i);
                    answer.insert(nums.end(), j);
                    cout << nums[i];
                    cout << nums[j];
                    return answer;
                }
            }
        }
        return answer;
    }
};

File: 1/JS/1.js
// /**
//  * @param {number[]} nums
//  * @param {number} target
//  * @return {number[]}
//  */

//First Answer, not as fast as second answer
//  var twoSum = function(nums, target) {
//     for (let i = 0; i < nums.length; i++) {
//         for (let j = i+1; j < nums.length; j++) {
//             if (nums[i] + nums[j] == target) {
//                 var resultArray = new Array(i, j);
//                 return resultArray;
//             }
//         }
//     }
// };

//Final Answer
var twoSum = function(nums, target) {
    let newArray = {};
    for (let i = 0; i < nums.length; i++) {
        let num = nums[i];
        let diff = target-num;
        if (newArray[diff] !== undefined) {
            let resultArray = new Array(newArray[diff], i);
            return resultArray;
        }
        newArray[num] = i;
    }
    return [];
};